### ISSUE TITLE
Cannot answer request on event 'onRequestEnd'

### ISSUE THREAD

**Issue by DblK** [joeferner/node-http-mitm-proxy#187]
Hi,

I try to match some query to do some specific answer. Until now, I was using the event `onRequest` which was working fine until I wanted to handle body request.
After mapping `onRequestData` & `onRequestEnd` I finally have all information need.
However, I cannot respond to the client. It seems somewhere in the library, a response is sent to the client.
Could you pinpoint where it is and how to avoid it? Or maybe to have all body request in `onRequest` event?

Thank you

**Comment by DblK** [comment]
To complete my issue, I know have the body of the request but the connection to the server is established during `onRequestHeaders`. At this point I wish to have the server connection but only after the `requestEnd` to be able to handle the body and then maybe or maybe not make connection to server.

Any idea?

**Comment by kics223w1** [comment]
Hi, did you solve the problem 🤔 I'll very appreciate if you share with me any idea @DblK

**Comment by DblK** [comment]
@kics223w1 yes I did solve my issue, making my own library [yamp](https://github.com/DblK/yamp).

**Comment by kics223w1** [comment]
- This is [my custom node-http-mitm-proxy](https://www.npmjs.com/package/custom-node-http-mitm-proxy) library. 
 - This library allows you can get the request body before the `onRequest` event is triggered.
 - I.e. If the body doesn't meet your condition, you can prevent it to proceed to the real server.
